src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -113,6 +113,9 @@ const InfiniteScroll = ({
         ),
       );
 
+      // Do not update the next if both are same
+      if (nextBeginPage === nextEndPage) return;
```

Comment:
Suppose the pageHeight is (2 * height), beginPage is 0, and endPage is 1. As the user scrolls down, there is a point in time where nextBeginPage will be 1 and nextEndPage will be 1. At this point, we should be updating beginPage to be 1 and leaving endPage at 1. With the proposed change above, that will not happen and beginPage will remain at 0.

I am trying to understand what the issue is. Can you provide a code example that demonstrates the issue?

Reply:
I could be in the wrong direction. 

The issue is happening in my specific case, where the search drop down in inside a scrollable modal. In my case the `items` array of size 6 and step defaulting at 20, Initially, `beginPage` is 0 and `endPage` is at 1 and loads fine, InfiniteScroll triggers an update and now the `beginPage` & `endPage` are arrived to be 1, due to this the items slicing goes for a toss on line below because `firstIndex` goes outside boundary array and results in them being empty.

https://github.com/grommet/grommet/blob/626f1e345d1aa3096748ce7c96180dfb96f2e61f/src/js/components/InfiniteScroll/InfiniteScroll.js#L168

Reply:
@ericsoderberghp  Here's a minimum reproduction on [codesandbox](https://codesandbox.io/s/dreamy-elgamal-obbhm?fontsize=14&hidenavigation=1&theme=dark) - https://obbhm.csb.app/

On the full desktop view
 - Without scrolling, click the toggle on right corner, now the options show up properly
 - Now hide layer, scroll the document to bottom, and toggle the layer again. This time, the options disappear.

This is happening because InfiniteScroll looks upto document as its scroll parent and hence the mismatch. Just curious, wouldn't `beginPage` being same as `lastPage` result in wrong slice of array?

Reply:
@ericsoderberghp  Any update over this?

Reply:
Apologies for the way too long delay on this. I just tried the codesandbox link provided above and couldn't seem to reproduce it. I couldn't scroll anywhere. I will try to reproduce still, just wanted to comment on the sandbox.